theano/sandbox/cuda/dnn.py

```diff
@@ -368,6 +368,27 @@ def ensure_float(val, default, name):
     return val
 
 
+def do_merge_gpu_alloc_empty(node1, node2):
```

Comment:
Can you explain the logic in words here? I don't understand what we are trying to do.